[vboxwrapper] Build is failed on the older system because of c++11 lambda usage

https://github.com/BOINC/boinc/blob/213028bb9523ee14800ad0866202f28945c3792c/samples/vboxwrapper/vbox_vboxmanage.cpp#L608-L609

This need to be rewritten to get rid of:
- lambda
- cbegin/cend

@computezrmle, please fix that

I get this error on the Mac:
`vbox_vboxmanage.cpp:609:44: error: expected expression
                        type_line.begin(), [](unsigned char c) { return tolower(c); });
`

--- a/samples/vboxwrapper/vbox_vboxmanage.cpp
+++ b/samples/vboxwrapper/vbox_vboxmanage.cpp
@@ -605,9 +605,7 @@ namespace vboxmanage {
                     //
 
                     type_line = output;
-                    transform(type_line.cbegin(), type_line.cend(),
-                        type_line.begin(), [](unsigned char c) { return tolower(c); });
-                    type_start = type_line.find("\ntype: ") + 1;
+                    type_start = type_line.find("\nType: ") + 1;
                     type_end   = type_line.find("\n", type_start) - type_start;
                     type_line  = type_line.substr(type_start, type_end);
 

